src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -389,13 +389,18 @@ static IEnumerable<Instruction> CompileBindingPath(ElementNode node, ILContext c
 				n = n.Parent as IElementNode;
 			}
 
-			if (dataTypeNode is null)
+			if (dataTypeNode is null) {
+				context.LoggingHelper.LogWarningOrError(10101, context.XamlFilePath, node.LineNumber, node.LinePosition, 0, 0, $"Binding could be compiled if x:DataType is specified", null);
 				yield break;
+			}
 
 			if (dataTypeNode is ElementNode enode
 				&& enode.XmlType.NamespaceUri == XamlParser.X2009Uri
 				&& enode.XmlType.Name == nameof(Microsoft.Maui.Controls.Xaml.NullExtension))
+			{
+				context.LoggingHelper.LogWarningOrError(10102, context.XamlFilePath, node.LineNumber, node.LinePosition, 0, 0, $"Binding could be compiled if x:DataType is not explicitly null", null);
```

Comment:
Should these text messages be resources ? So they can be localized ?

Reply:
that's a good point. I'm never in for localizing runtime errors, but compile time are fine